### PR TITLE
Create CONTRIBUTING.md

### LINKED ISSUE #1435: Create a CONTRIBUTING.md
@michaelshobbs I think it's time we codify how we merge stuff, decide what goes into a release, how people should contribute etc. Do you have thoughts on what is useful data to provide for bugs? What about the structure of how and why people should ask for features vs things that should go into plugins?

We'll also want to extend this to the official plugins, though some of the rules may be different.

/cc @Flink since you're starting to get back into contributing and might have some thoughts as an outsider.

### PR DESCRIPTION
**Pull request by josegonzalez**
Closes #1435

### CODE REVIEW

**Comment by josegonzalez** [comment]
I think we can start here and iterate as we see what works?

**Comment by Flink** [comment]
From https://github.com/docker/docker/blob/master/CONTRIBUTING.md#quick-contribution-tips-and-guidelines there’s also a little section on git and I think we should incite good practices when writing commits like using rebase, squash commits, etc. (starting with us ;))

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub.
+
+### Making Changes
+
+* Create a topic branch from where you want to base your work.
+  * This is usually the master branch.
+  * Only target an existing branche if you are certain your fix must be on that
```
Comment (michaelshobbs) [inline CONTRIBUTING.md]: branche?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    branch.
+  * To quickly create a topic branch based on master; `git branch
+    master/my_contribution master` then checkout the new branch with `git
+    checkout master/my_contribution`. Better avoid working directly on the
+    `master` branch, to avoid conflicts if you pull in updates from origin.
```
Comment (michaelshobbs) [inline CONTRIBUTING.md]: maybe reference `git checkout -b` as a single step instead of branch and checkout

**Comment by michaelshobbs** [comment]
:+1: on @Flink's comment

File: CONTRIBUTING.md (hunk 1 above)
Comment (Flink) [inline CONTRIBUTING.md]: that’s french for branch :grin:

**Comment by josegonzalez** [comment]
Updated the `Making Changes` section to take @Flink's comment into account. Merging :)

**Comment by Flink** [comment]
Nice :+1:

**Pull request merged**